src/common/models/Projections.ts

```diff
@@ -62,41 +63,41 @@ export class Projections {
     return this.projected;
   }
 
+  getMetricValue(metric: Metric): number | null {
+    if (!this.primary) {
+      return null;
+    }
+    switch (metric) {
+      case Metric.CASE_GROWTH_RATE:
+        return this.primary.rt;
+      case Metric.HOSPITAL_USAGE:
+        return this.primary.currentIcuUtilization;
+      case Metric.POSITIVE_TESTS:
+        return this.primary.currentTestPositiveRate;
+      case Metric.CONTACT_TRACING:
+        return this.primary.currentContactTracerMetric;
+      default:
+        fail('Cannot get value of metric: ' + metric);
+    }
+  }
+
+  getMetricLevel(metric: Metric): Level {
+    return getLevel(metric, this.getMetricValue(metric));
+  }
+
+  // TODO(michael): Rework this to return a { [metric in Metric]: Level } map
+  // instead of using the custom "rt_level", etc. keys.
   getLevels(): {
     rt_level: Level;
     hospitalizations_level: Level;
     test_rate_level: Level;
     contact_tracing_level: Level;
   } {
-    const projection = this.primary;
-    if (!projection)
-      return {
-        rt_level: Level.UNKNOWN,
-        hospitalizations_level: Level.UNKNOWN,
-        test_rate_level: Level.UNKNOWN,
-        contact_tracing_level: Level.UNKNOWN,
-      };
-
```

Comment:
hmm don't we still need this case?

Reply:
GetMetricValue() returns `null` if `projections.primary` is null, and `getLevel()` returns UNKNOWN if you pass a `null` metric value, so we don't need to special case this `!projection` case anymore.

Reply:
cool cool